gnovm/pkg/gnolang/ownership.go

```diff
@@ -359,6 +362,9 @@ func (tv *TypedValue) GetFirstObject(store Store) Object {
 		oo := store.GetObject(cv.ObjectID)
 		tv.V = oo
 		return oo
+	case *HeapItemValue:
+		// should only appear in PointerValue.Base
+		panic("should not happen")
```

Comment:
HeapItemValue never appears otherwise, except as Base to PointerValue. 
This defensive check should help in the future.